stream_alert/alert_processor/outputs.py

```diff
@@ -803,3 +805,272 @@ def dispatch(self, **kwargs):
+
+        comment_url = os.path.join(self.base_url, self.COMMENT.format(issue_id))
+        comment_body = {'body' : comment}
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
+        resp = self._post_request(comment_url, data=comment_body, headers=headers, verify=False)
+        jira_id = False
+
+        try:
+            jira_id = resp.json()['id']
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while creating Jira comment [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+        return jira_id
+
+
+    def _get_comments(self, issue_id):
+        """Get all comments for an existing Jira issue
+
+        Args:
+            issue_id (str): The existing issue ID or key
+
+        Returns:
+            list: List of comments associated with a Jira issue
+        """
+        comment_url = os.path.join(self.base_url, self.COMMENT.format(issue_id))
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
+        resp = self._get_request(comment_url, headers=headers, verify=False)
+        comments = []
+
+        try:
+            comments = resp.json()['comments']
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while getting Jira comments [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+        return comments
+
+    def _get_existing_issue(self, issue_summary, project_key):
+        """Find an existing Jira issue based on the issue summary
+
+        Args:
+            issue_summary (str): The Jira issue summary
+            project_key (str): The Jira project to search
+
+        Returns:
+            int: ID of the found issue or False if existing issue does not exist
+        """
+        jql = 'summary ~ "{}" and project="{}"'.format(issue_summary, project_key)
+        resp = self._search_jira(jql, fields=['id', 'summary'], max_results=1)
+        jira_id = False
+
+        try:
+            jira_id = resp[0]['id']
+        except IndexError:
+            LOGGER.debug('Existing Jira issue not found')
+
+        return jira_id
+
+    def _create_issue(self, issue_name, project_key, issue_type, description):
+        """Create a Jira issue to write alerts to. Alert is written to the "description"
+        field of an issue.
+
+        Args:
+            issue_name (str): The name of the Jira issue
+            project_key (str): The Jira project key which issues will be associated with
+            issue_type (str): The type of issue being created
+            description (str): The body of text which describes the issue
+
+        Returns:
+            int: ID of the created issue or False if unsuccessful
+        """
+        issue_url = os.path.join(self.base_url, self.ISSUE)
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
+        issue_body = {'fields':{}}
```

Comment:
It would be okay, and probably a little more readable, to create this object all in one go, like:
```
issue_body = {
    'fields': {
        'project': {
            'key': project_key
        },
        'summary': issue_name,
        'description': description,
        'issuetype': {
            'name': issue_type
        }
    }
}
```

Reply:
I originally did define the dict like this but pylint was throwing a fit.  I must have screwed up the indentation or something as it seems to work just fine now.